File: CodingSamples/Java/Language/InterfaceTest1/src/CurrentAccount.java
package banking;

//a class defined with final modifier cannot be extended
final class CurrentAccount extends Account {

	public void withdraw(double amount) throws InsufficientFundsException {
		balance -= amount;
	}

	public void deposit(double amount) {
		if(balance < 0)
			amount *= 0.9;
		balance += amount;
	}
}

File: CodingSamples/DotNet/Language/InterfaceTest1/BankLib/CurrentAccount.cs
namespace Banking;

//a class declared with sealed modifier cannot
//be used as a base class
sealed class CurrentAccount : Account, IFineable
{
    public override void Deposit(double amount)
    {
        Balance += amount;        
    }

    public override void Withdraw(double amount)
    {
        Balance -= amount;
    }

    //explicit implementation of IFineable
    bool IFineable.Withdraw(double fine)
    {
        if(Balance < 0)
        {
            Balance -= fine;
            return true;
        }
        return false;
    }
}
